go-controller/pkg/util/ipsec.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func GetAllEstablishedIPsecTunnels() (sets.Set[string], error) {
```

Comment:
Is there an ipsec client to get this info ?

Reply:
AFAIK we have only `ipsec` cli to connect with pluto daemon, do you see any problem with using it ?